infra/cifuzz/fuzz_target.py

```diff
@@ -121,6 +124,32 @@ def is_reproducible(self, test_case):
         return True
     return False
 
+  def is_crash_valid(self, test_case):
+    """Checks if a crash was introduced by the pull request.
+
+    NOTE: If no old_build_target is specified the crash is assumed introduced
+    by the pull request.
+
+    Args:
+      test_case: The path to the test_case that triggered the crash.
+
+    Returns:
+      True if the crash was introduced by the current pull request.
+    """
+    if not self.old_build_path:
+      return True
+    exists_in_pr = self.is_reproducible(test_case,
+                                        os.path.dirname(self.target_path))
+    if not exists_in_pr:
+      logging.info('Crash is not reproducible.')
+      return False
+    exists_in_master = self.is_reproducible(test_case, self.old_build_path)
+    if exists_in_pr and not exists_in_master:
+      logging.info('Crash is new and reproducible.')
+      return True
+    logging.info('Crash was found in old OSS-Fuzz build.')
+    return False
```

Comment:
Unless I'm mistaken, this kind of makes it harder to use CIFuzz to make sure PRs that are supposed to fix bugs discovered by OSS-Fuzz really fix them. Is there going to be a way to prevent CIFuzz from ignoring some crashes?

Reply:
I haven't looked at the PR where CIFuzz has started downloading corpora from GCP yet but I think if it takes the "*_seed_corpus.zip" files into account it should be possible not to ignore crashes that can be triggered by those "local" corpora. This way, it seems it's going to be relatively easy for developers to let CIFuzz know what it shouldn't ignore by adding testcases for bugs they are fixing to *_seed_corpus.zip. I hope it makes sense.

Reply:
>Unless I'm mistaken, this kind of makes it harder to use CIFuzz to make sure PRs that are supposed to fix bugs discovered by OSS-Fuzz really fix them.

I think you're not mistaken, it would make this harder.

I didn't think of this use case when we came up with this feature.
How important is this use case to you? I guess it might be nice for devs in projects that use OSS-Fuzz but who don't really know it.
I think the related case of an incomplete fix (a slight variation on it finds essentially the same crash) is interesting.

I'll think about how we could support this case.
Maybe we could use a [trigger word](https://github.com/Khan/pull-request-comment-trigger) so that CIFuzz will skip this crash suppression? Like if user comments on a PR or includes a certain string in the description we won't suppress crashes.

>This way, it seems it's going to be relatively easy for developers to let CIFuzz know what it shouldn't ignore by adding testcases for bugs they are fixing to *_seed_corpus.zip. I hope it makes sense.

I think I understand this. Basically don't suppress crashes if they are in the seed corpus (or added to it in a fancier version).
I like the simplicity of this but I'm worried about changing behavior like this behind the scenes  without the user knowing, so maybe a trigger word is better. I wonder if its safe to assume that users will never want to add a crashing item to their corpus without a fix.
Trigger could also handle incomplete fixes better.

Need to think more about this.

Reply:
> How important is this use case to you?

I wouldn't say it's very important to me at this point mostly because there is another (systemd-specific) way (which I hope eventually will be retired) to automatically make sure PRs addressing OSS-Fuzz issues really fix them but I think it would be helpful in general in the sense that usually it's expected that CI can be used to verify patches on the fly by simply adding tests (or testcases).

> I like the simplicity of this but I'm worried about changing behavior like this behind the scenes without the user knowing

I agree.

> Maybe we could use a trigger word so that CIFuzz will skip this crash suppression? Like if user comments on a PR or includes a certain string in the description we won't suppress crashes.

As far as I understand, with trigger words it will be necessary to add both a testcase/testcases and a trigger word even though the presence of the testcase should be enough to let CI know it shouldn't ignore it. Generally I'm not against "trigger words" (and probably some projects would prefer to use them) but I think it would be great if I could put testcases in "$OUT/fuzz-target_should_not_crash.zip" so that CIFuzz would pick them up and always run the fuzz targets against them and report back if any of them trigger anything.

Reply:
> I think it would be great if I could put testcases in "$OUT/fuzz-target_should_not_crash.zip"

On second thought, it seems to be too kludgy because for this to work I'd have to tweak the build script so that it could detect CIFuzz (probably by looking for some environment variable that isn't there yet :-) and put files kept in `systemd/test/fuzz/*/` in `$OUT`. In theory, it should be possible to specify where CIFuzz can find testcases that shouldn't crash fuzz targets but having looked at the way a couple of projects integrated into OSS-Fuzz fix bugs it looks like PRs are merged based on code review only. I'm not sure hoping for the best (and relying on ClusterFuzz closing (or not) issues automatically eventually)) is the best strategy but it seems to work fine for some (probably most?) projects.